### PR TITLE
Salt handler

### LINKED ISSUE #93: [Backend | Login & Auth API] Get Salt Handler & Use Case

### PR DESCRIPTION
**Pull request by Jacob4649**
Closes #93.

### CODE REVIEW

**Review by Enkhzul001 — APPROVED** [review]
Everything looks good.

**Review by Enkhzul001 — APPROVED** [review]
Looks good.

File: serverless.yml [hunk 1]
```diff
@@ -40,6 +40,13 @@ functions:
           path: /login
           method: post
 
+  salt:
+    handler: com.javahelp.backend.endpoint.login.SaltHandler
+    events:
+      - httpApi:
+          path: /users/salt
+          method: get
+
```
Comment (Lenzs) [inline serverless.yml:49]: would it be better to pass in userid as a path param and have a separate path to fetch the salt via userid? something like /users/{userid}/salt?

File: backend/src/main/java/com/javahelp/backend/endpoint/login/SaltHandler.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    .setStatusCode(OK)
+                    .build();
+        } else {
+            String json = Json.createObjectBuilder()
+                    .add("errorMessage", result.getErrorMessage())
+                    .build().toString();
+            return APIGatewayResponse.builder()
+                    .setJSONBody(json)
+                    .setStatusCode(OK)
+                    .build();
```
[the remark this answers is not in the document]
Comment (Jacob4649) [reply backend/src/main/java/com/javahelp/backend/endpoint/login/SaltHandler.java:93]: Good point, I'll change it

File: serverless.yml (hunk 1 above)
Comment (Jacob4649) [reply serverless.yml:49]: Another good point, ill add a new route

File: backend/src/main/java/com/javahelp/backend/endpoint/APIGatewayV2ProxyRequestEvent.java [hunk 3]
```diff
@@ -81,7 +103,20 @@ public Map<String, String> getHeaders() {
...
+     */
+    public Map<String, String> getPathParameters() {
+        return pathParameters;
+    }
+
+    /**
+     * @return the {@link String} raw path for this request
+     */
+    public String getRawPath() {
+        return rawPath;
```
[the remark this answers is not in the document]
Comment (Jacob4649) [reply backend/src/main/java/com/javahelp/backend/endpoint/APIGatewayV2ProxyRequestEvent.java:119]: im gonna leave them in to implement the second route for when going by user id

**Review by BullDF — APPROVED** [review]

**Review by Enkhzul001 — APPROVED** [review]
looks good.

**Review by BullDF — APPROVED** [review]
Good stuff

**Review by Lam207 — APPROVED** [review]
Great work! Thanks Billy for pointing some out

**Pull request merged**